app/controllers/DataRetrievals.scala

```diff
@@ -46,6 +45,14 @@ object DataRetrievals {
     }
   }
 
+  def retrievePSTRAndChargeADetails(block: (String, ChargeDetails) => Future[Result])
```

Comment:
As this is specific to charge A and is only used on charge A's CYA controller it perhaps is not necessary to have this method on this trait, instead I'd just put it on the CYA controller and make it private.

Reply:
Done